src/core/Elsa.Core/Handlers/PersistWorkflow.cs

```diff
@@ -28,12 +29,16 @@ public PersistWorkflow(IWorkflowInstanceStore workflowInstanceStore, ILogger<Per
 
         public async Task Handle(WorkflowSuspended notification, CancellationToken cancellationToken)
         {
+            _logger.LogTrace("Received notification {notification}", notification);
```

Comment:
I can see how logging is helpful. So how about we introduce a logging behavior for Mediatr? Something like what is described in [this blog post](https://dev.to/raulismasiukas/logging-pipeline-with-mediatr-5d4f). Then we can remove the repetitive logging statements, while automatically getting logging for all notification handlers.

Reply:
That seems pretty cool and I did wonder at the time whether I could do that logging better.  It wasn't my chief concern though because I only wanted some proof that the persistence methods were actually being used.

The generalised solution in the blog looks good.  The only thing I think which might be useful to add to that would be to record which handler method(s) are receiving the notifications.  I put the logging there to try and figure out "did this even get called?" (I'd have used a breakpoint but Omnisharp's debugger kept crashing on me when executing this code).  So, just "mediatr received a notification of type X and relayed it to zero or more handlers" doesn't tell me "Handler X was notified & Handler Y was notified but Handler Z was not" etc.

I'll see if I can adapt the example in the blog though, to a degree I'd forgotten I even left the logging logic in the PR.  But now it's in there I may as well do it properly 😄

Reply:
> I'll see if I can adapt the example in the blog though, to a degree I'd forgotten I even left the logging logic in the PR. But now it's in there I may as well do it properly 😄

Sounds good! But feel free to do it as a separate PR if you wish.

Reply:
I backed-out the repeated logging.  I had a look into the technique linked in the blog post but [it is no good for notification handlers, only request handlers](https://github.com/jbogard/MediatR/wiki/Behaviors#nb).

I went on to look into [the technique described here, adding custom publish behaviour](https://github.com/jbogard/MediatR/wiki#publish-strategies) - overriding `PublishCore` to inject logging.  It worked to a degree but clearly the instances of `Func<INotification, CancellationToken, Task>` which are received by this method do not point to _the real/original methods_ on the notification handler classes, they are lambda functions which wrap those original methods.  That means that [the `Target`  and `Method` properties](https://docs.microsoft.com/en-us/dotnet/api/system.delegate?view=net-5.0#properties) of that delegate provide only meaningless names, and don't point to the actual concrete class which was going to handle the notification.

Since this is an aside from the overall aim of this PR, I've just backed it all out.  The net effect is that I think there's not much (if any) change now.